### ISSUE TITLE
Spawn processes outside of the training loop

### ISSUE THREAD

**Issue by atgambardella** [atgambardella/pytorch-es#1]
As the code is now, `n` processes are spawned per gradient step. As python startup time takes a while (~30 ms per process), this causes non-negligible overhead.
